### PR TITLE
 LTD-2590-SIEL-non-firearms-establish-category 

### PR DESCRIPTION
**Pull request by depsiatwal**
I need to do more FF at some point to disable the 3 options in category. 
Will do this in follow up PRs when needed. For now we have 1 main one to disable the whole journey.

### CODE REVIEW

File: exporter/applications/views/goods/goods.py [hunk 1]
```diff
@@ -237,6 +248,37 @@ def reset(self):
...
+    template_name = "core/form.html"
+    form_class = NonFirearmCategoryForm
+
+    def get_success_url(self):
+        return reverse("applications:new_good_firearm", kwargs={"pk": self.kwargs["pk"]})
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["back_link_url"] = reverse("applications:goods", kwargs={"pk": self.kwargs["pk"]})
+        return context
```
Comment (kevincarrogan) [inline exporter/applications/views/goods/goods.py:279]: Is this waiting for future tickets for this to actually do anything useful?

File: exporter/applications/views/goods/goods.py [hunk 2]
```diff
@@ -237,6 +248,37 @@ def reset(self):
             super().reset()
 
 
+class IsGoodFirearm(LoginRequiredMixin, CheckNonFirearmEnabledMixin, FormView):
+    template_name = "core/form.html"
+    form_class = IsFirearmForm
+
+    def form_valid(self, form):
+        if str_to_bool(form.data["is_firearm_product"]):
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply exporter/applications/views/goods/goods.py:256]: yeah that's what I thought but then seen on other forms and assumed I needed 
seems to be working now so removed

[the remark this answers is not in the document]
Comment (depsiatwal) [reply exporter/applications/views/goods/goods.py:256]: that's what I thought too but seen on other forms as value seems to come through as string.

File: exporter/goods/forms/goods.py [hunk 3]
```diff
@@ -682,6 +686,67 @@ def __init__(self, *args, **kwargs):
...
+        COMPLETE_PRODUCT = "COMPLETE_PRODUCT", "It's a complete product"
+        PART_PRODUCT = "PART_PRODUCT", "It forms part of a product"
+        HELPS_PRODUCT = "HELPS_PRODUCT", "It helps to operate a product"
+
+    CATEGORY_CHOICES = (
+        Choice(
+            NonFirearmCategoryChoices.COMPLETE_PRODUCT.name,
+            NonFirearmCategoryChoices.COMPLETE_PRODUCT.label,
+            hint="Hardware such as devices, systems, platforms, vehicles, equipment.",
+        ),
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply exporter/goods/forms/goods.py]: Changed

File: exporter/goods/forms/goods.py [hunk 4]
```diff
@@ -682,6 +686,67 @@ def __init__(self, *args, **kwargs):
...
+        Choice(
+            NonFirearmCategoryChoices.PART_PRODUCT.name,
+            NonFirearmCategoryChoices.PART_PRODUCT.label,
+            hint="Hardware such as components and accessories, or raw materials and substances.",
+        ),
+        Choice(
+            NonFirearmCategoryChoices.HELPS_PRODUCT.name,
+            NonFirearmCategoryChoices.HELPS_PRODUCT.label,
+            hint="For example, software or information such as technology manuals and specifications.",
+            enable=False,
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply exporter/goods/forms/goods.py]: ahh left over from me playing around lol

File: unit_tests/exporter/applications/views/test_add_good_non_firearm_add.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+from django.urls import reverse
+from django.conf import settings
+import uuid
+
+
+@pytest.fixture(autouse=True)
+def setup():
+    settings.FEATURE_FLAG_NON_FIREARMS_ENABLED = True
+    pass
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply unit_tests/exporter/applications/views/test_add_good_non_firearm_add.py]: removed

File: unit_tests/exporter/applications/views/test_add_good_non_firearm_add.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+
+@pytest.fixture(autouse=True)
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply unit_tests/exporter/applications/views/test_add_good_non_firearm_add.py]: changed

File: unit_tests/exporter/applications/views/test_add_good_non_firearm_add.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    (
+        "new_good_firearm",
+        "non_firearm_category",
+    ),
+)
+def test_is_good_firearm_view_raise_404(authorized_client, application_pk, post_url):
+    settings.FEATURE_FLAG_NON_FIREARMS_ENABLED = False
+    url = reverse(f"applications:{post_url}", kwargs={"pk": application_pk})
+    response = authorized_client.get(url)
+    response.status_code = 404
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply unit_tests/exporter/applications/views/test_add_good_non_firearm_add.py]: changed

File: exporter/applications/views/goods/goods.py (hunk 1 above)
Comment (depsiatwal) [reply exporter/applications/views/goods/goods.py:279]: correct the flow will come together once other tickets are in play 
in fact this takes you back to goods list

File: exporter/goods/forms/goods.py [hunk 8]
```diff
@@ -682,6 +686,67 @@ def __init__(self, *args, **kwargs):
         )
 
 
+class IsFirearmForm(BaseForm):
+    class Layout:
+        TITLE = "Is it a firearm product?"
+
+    is_firearm_product = forms.TypedChoiceField(
```
[the remark this answers is not in the document]
Comment (depsiatwal) [reply exporter/goods/forms/goods.py:694]: implemented and refactored a little

File: exporter/applications/views/goods/goods.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevincarrogan) [reply exporter/applications/views/goods/goods.py:256]: Don't take those forms as the precedent for this, it's not the right thing to do.

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**